apps/web/lib/slots.ts

```diff
@@ -15,26 +15,34 @@ export type GetSlots = {
   frequency: number;
   workingHours: WorkingHours[];
   minimumBookingNotice: number;
+  eventLength: number;
 };
 export type WorkingHoursTimeFrame = { startTime: number; endTime: number };
 
 const splitAvailableTime = (
   startTimeMinutes: number,
   endTimeMinutes: number,
-  frequency: number
+  frequency: number,
+  eventLength: number
```

Comment:
`eventLength` can be different from `frequency` when custom slot intervals are used, and so, we need eventLength to ensure the final few slots don't allow their respective end time to be beyond the user's availability end time.